### DISCUSSION TITLE
Mouse cursor manipulation through ssh

### DISCUSSION THREAD

**Discussion by infogrind** [kovidgoyal/kitty#5220] [General]
I really like Kitty's shell integration feature that allows me to place the cursor using the mouse. I've noticed that this is not possible through SSH (I'm using the SSH kitten). Is there any way to enable this, or is future support planned?

**Comment by kovidgoyal** [comment]
works fine through ssh with the ssh kitten. Make sure the shell you are running on the remote server is a reasonably recent one.

**Reply by infogrind** [reply]
Thanks! What does "reasonably recent one" mean? I have Fish 3.4.1 both locally and remotely. When connected via SSH, the cursor is a block, not a vertical line.

And I really appreciate the fast answer, great support!

**Reply by page-down** [reply]
Use `kitty +kitten ssh <YOUR_HOST>` to connect with SSH, or manually install shell integration to the remote host.

https://sw.kovidgoyal.net/kitty/kittens/ssh/
https://sw.kovidgoyal.net/kitty/shell-integration/

**Reply by infogrind** [reply]
Thanks! I am already using “kitty +kitten ssh” to connect (see my initial question).

**Comment by kovidgoyal** [comment]
On Wed, Jun 22, 2022 at 12:19:31AM -0700, Marius Kleiner wrote:
> Thanks! What does "reasonably recent one" mean? I have Fish 3.4.1 both locally and remotely. When connected via SSH, the cursor is a block, not a vertical line.

Not bash 3.2 :) fish 3.4 should be fine. Is shell integration working
for you at all when sshed? run ls and press shift+ctrl+g

**Reply by infogrind** [reply]
When I run `ls` and press `shift+ctrl+g`, `less` is opening. It is opening with the entire scrollback (all past commands), rather than just the output of `ls`.

**Comment by kovidgoyal** [comment]
On Wed, Jun 22, 2022 at 02:22:05AM -0700, Marius Kleiner wrote:
> When I run `ls` and press `shift+ctrl+g`, `less` is opening. It is opening with the entire scrollback (all past commands), rather than just the output of `ls`.

That indicates shell integration is broken on the server. @page-down
fish integration is your code you want to have a look? @Marius try using
an empty fish config on the server

**Reply by page-down** [reply]
I specifically tested Tide (fish prompt) a few months ago because of shell integration and had some issues at the time.
I just tried it again on purpose (maybe OP is using this) when I saw this discussion, and the cursor works fine, the prompt marking feature works fine, CWD reports fine, and clone-in-kitty works fine.

To check shell integration, then manually configure it for troubleshooting first.
To check ssh kitten, first check if the file is successfully transferred in the default data location, then try to explicitly enable shell integration in the configuration file.
It will be easier to troubleshoot the problem starting with an empty configuration.

**Comment by kovidgoyal** [comment]
Testing with

```
kitty +kitten ssh --kitten=login_shell=fish localhost
```

works fine for me.

**Reply by infogrind** [reply]
I think I know what is wrong. I am using this in my company setup, which has some special constraints which mean I can't default my login shell on my remote server to `fish` but have to keep it set to `bash` and run `fish` via `.bashrc`.

If I use the `--kitten=login_shell=fish`, it works great! So I'll just use that in the future.

Thanks again.
